### ISSUE TITLE
Android Firebase Crashlytics reporting ANR's (Application Not Responsive) errors coming from LD RN SDK 6.2.0

### ISSUE THREAD

**Issue by fparhat-rbi** [launchdarkly/react-native-client-sdk#144]
**Describe the bug**

- We are successfully consuming the `6.2.0` SDK in our Expo RN application but crashlytics is reporting ANR errors for a variety of our android users. (See Logs)

**To reproduce**
- Unknown

**Expected behavior**
- ANR's should not be occurring from the Android SDK

**Logs**
If applicable, add any log output related to your problem.

![image](https://user-images.githubusercontent.com/91911432/178348366-f09f2c97-1aa5-4109-b683-8734f980c60b.png)
![image](https://user-images.githubusercontent.com/91911432/178348674-da0ca3fb-e8e4-4d9e-9e7c-75c6dfec7645.png)
```
java.lang.Object.wait (Native method)
java.util.concurrent.TimeUnit.timedWait (TimeUnit.java:353)
com.launchdarkly.sdk.android.LDAwaitFuture.get (LDFutures.java:144)
com.launchdarkly.sdk.android.LDClient.init (LDClient.java:196)
com.launchdarkly.reactnative.LaunchdarklyReactNativeClientModule$1.run (LaunchdarklyReactNativeClientModule.java:243)
```

Main thread (blocked)
```
com.launchdarkly.sdk.android.LDClient.onNetworkConnectivityChangeInstances (LDClient.java)
com.launchdarkly.sdk.android.ConnectivityReceiver.onReceive (ConnectivityReceiver.java:29)
android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args (LoadedApk.java:1806)
```

**SDK version**
6.2.0

**Language version, developer tools**
Package.json
```
    "launchdarkly-react-native-client-sdk": "6.2.0",
    "expo": "~45.0.0",
    "react": "17.0.2",
    "react-native": "0.68.2",
```

**Comment by fparhat-rbi** [comment]
It looks like this issue on the Android SDK is the same,
https://github.com/launchdarkly/android-client-sdk/issues/142

As we are seeing the main thread being blocked on `com.launchdarkly.sdk.android.LDClient.onNetworkConnectivityChangeInstances (LDClient.java)`

![CleanShot 2022-07-11 at 16 28 23](https://user-images.githubusercontent.com/91911432/178352471-cfe51225-2a87-453a-9d4f-9bea9d9a38a9.png)

**Comment by kinyoklion** [comment]
Hello @fparhat-rbi,

Thank you for these great stacks. We believe the other issue you listed does have the same root cause. There is a synchronized blocking version of the `init` method in the android SDK, which the RN SDK uses, even though this work is offloaded to another thread in RN, it will still cause that connectivity handler to be blocked on the main thread until it completes. Similarly, if using the android SDK directly, and not using the futures version of the init method, the same problem could be encountered.

We are reviewing this area of code to come up with a solution that should handle a number of these issues.

Thank you,
Ryan

**Comment by louis-launchdarkly** [comment]
Hello @fparhat-rbi, we have just released the React Native SDK version 6.2.1 that has the change to resolve the `init/onNetworkConnectivityChangeInstances` related ANR. Please try it out and let us know how it goes. If you see any issues, please feel free to reach out to LaunchDarkly Support, or open a new issue.
